homeassistant/components/sensor/__init__.py

```diff
@@ -508,67 +565,85 @@ def state(self) -> Any:
                 )
             return value
 
+        numerical_value: int | float | Decimal | None = None
+        numerical_conversion_failed = None
+        if not isinstance(value, (int, float, Decimal)):
```

Comment:
Is there a reason to try to convert to a number if `(prec := self.precision) is None`?

I think it may be better to adjust the early return on (old) line 513:
```python
        # If it doesn't need to be a number, return existing value
        if not device_class and not state_class and not unit_of_measurement and (prec := self.precision) is None:
            return value
```

And move the conversion after the check.

Reply:
Good call, fixed 👍

Reply:
almost... I think you need to add `and (prec := self.precision) is None` to the check above :)